File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        values = []
        curr = head
        while curr:
            values.append(curr.val)
            curr = curr.next
        
        twinSum = 0
        i = 0
        j = len(values) - 1
        while i < j:
            twinSum = max(twinSum, values[i] + values[j])
            i += 1
            j -= 1
        return twinSum

File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {number}
 */

// n = 8
// 0 1 2 3 4 5 6 7
// 
// i = 0
// 8 - 1 - 1
// n - 1 - i
var pairSum = function(head) {
    let values = [];
    let curr = head;
    while (curr) {
        values.push(curr.val);
        curr = curr.next;
    }
    
    let maxTwinSum = 0;
    let i = 0, j = values.length - 1;
    while (i < j) {
        maxTwinSum = Math.max(maxTwinSum, values[i] + values[j]);
        i++;
        j--;
    }
    return maxTwinSum;
};